File: android/app/src/main/java/vn/team/freechat/constant/MessageType.java
package vn.team.freechat.constant;

import com.tvd12.ezyfoxserver.client.constant.EzyConstant;

/**
 * Created by tavandung12 on 10/5/18.
 */

public enum MessageType implements EzyConstant {
    SENDING(1),
    SENT(2),
    RECEIVED(3);

    private final int id;

    private MessageType(int id) {
        this.id = id;
    }

    @Override
    public int getId() {
        return id;
    }

    @Override
    public String getName() {
        return toString();
    }
}

File: kotlin/app/src/main/java/vn/team/freechat_kotlin/constant/MessageType.kt
package vn.team.freechat_kotlin.constant

import com.tvd12.ezyfoxserver.client.constant.EzyConstant

/**
 * Created by tavandung12 on 10/5/18.
 */

enum class MessageType constructor(private val id: Int) : EzyConstant {
    SENDING(1),
    SENT(2),
    RECEIVED(3);

    override fun getId(): Int {
        return id
    }

    override fun getName(): String {
        return toString()
    }
}
